storage/oasis/nodeapi/file/kvstore.go

```diff
@@ -87,7 +89,115 @@ func (s pogrebKVStore) Close() error {
 	return s.db.Close()
 }
 
+// Returns true if path exists. Uses simplified error handling
+// to match pogreb's behavior.
+func pathExists(path string) bool {
+	_, err := os.Stat(path)
+	return err == nil
+}
+
+// Returns a list of files that match any of the patterns, but do not match any of the antipatterns.
+func glob(patterns []string, antipatterns []string) ([]string, error) {
+	files := map[string]struct{}{}
+	for _, pattern := range patterns {
+		matches, err := filepath.Glob(pattern)
+		if err != nil {
+			return nil, err
+		}
+		for _, match := range matches {
+			files[match] = struct{}{}
+		}
+	}
+	for _, antipattern := range antipatterns {
+		matches, err := filepath.Glob(antipattern)
+		if err != nil {
+			return nil, err
+		}
+		for _, match := range matches {
+			delete(files, match)
+		}
+	}
+	filesArr := make([]string, 0, len(files))
+	for k := range files {
+		filesArr = append(filesArr, k)
+	}
+	return filesArr, nil
+}
+
+// Moves all files that match the src glob patterns to the destination directory.
+func moveFiles(srcPatterns []string, srcAntipatters []string, dst string) error {
+	files, err := glob(srcPatterns, srcAntipatters)
+	if err != nil {
+		return fmt.Errorf("unable to glob for files to move: %w", err)
+	}
+
+	// Create the destination directory.
+	if err := os.MkdirAll(dst, 0o700); err != nil {
+		return fmt.Errorf("unable to create destination directory %s: %w", dst, err)
+	}
+
+	for _, srcFile := range files {
+		dstFile := filepath.Join(dst, filepath.Base(srcFile))
```

Comment:
Maybe worth adding a comment that this is safe to do because the glob pattern that is used in `preBackup` will only return objects from a single folder. Otherwise this could override objects with same name from different folders.

Reply:
Technically, I think `os.Rename()` will fail for the 2nd and all subsequent files with the same filename, and this func will return with an error. I added a warning in the func docstring though, it's a good caveat to call out.